### PR TITLE
Errors during prerender stop script

### PR DESCRIPTION
**Pull request by apepper**
Original patch by https://github.com/cbou (closes #363).

### CODE REVIEW

File: prerender/storePrerenderedContent.js [hunk 1]
```diff
@@ -86,6 +86,15 @@ function log(message, ...args) {
   console.log(`[storePrerenderedContent] ${message}`, ...args);
 }
 
+async function startBrowser() {
+  try {
+    return await puppeteer.launch();
+  } catch (e) {
+    reportError("Could not start browser!", e);
+    process.exit(1);
```
Comment (dcsaszar) [inline prerender/storePrerenderedContent.js:94]: IIRC, using `process.exit` may lead to loss of log output. So, this fix kinda defeats the purpose. We should use the same strategy as below, with `process.exitCode = 1;` and ending the script gracefully.

**Review by dcsaszar — CHANGES_REQUESTED** [review]
See comment. I hate this subtle `process.exit` issue, but let's try to avoid the pitfall.

File: prerender/storePrerenderedContent.js (hunk 1 above)
Comment (apepper) [reply prerender/storePrerenderedContent.js]: Good point.

**Comment by apepper** [comment]
REVIEW (11f9053)

I did a closer analysis of the error. Node would not exit the process with the original `process.exitCode = 1` implementation, since an express server would still run.

I forced pushed a new version, that will always close the browser and the server. This is regardless of errors in between. With that node should exit as expected.

**Review by dcsaszar — APPROVED** [review]
LGTM.

* [ ] Q: How does this change fix the original complaint "no useful log if the browser failed"?

**Comment by apepper** [comment]
> Q: How does this change fix the original complaint "no useful log if the browser failed"?

My guess is, that `AWSAmplify` only provided logs after a process was fully done. Since the process never terminated it did not output that message.

But I did not test this myself. I only tested it locally with a invalid `executablePath`.

**Comment by cbou** [comment]
Thanks for your PR! I think it's should be fine for AWS.

**Pull request merged**